src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -79,9 +77,13 @@ export function DownloadButton(props) {
 
     const onClick = useCallback(() => {
         if (selectInfoCls.getSelectedCount()) {
-            var panel = props.children ? React.Children.only(props.children) : <DownloadOptionPanel/>;
-            panel = React.cloneElement(panel, {tbl_id});
-            showDownloadDialog(panel);
+            if(hasOnlyProprietaryData(getTblById(tbl_id))){
+                showInfoPopup('You have chosen private data to download.', 'Private Data Selected');
```

Comment:
Please, before merging, change dialog sentence to “You do not have permission to download the selected data set(s).”

Reply:
Done!